### DISCUSSION TITLE
【主库】ObjectPool.Get() timeout 10 seconds

### DISCUSSION THREAD

**Discussion by 2881099** [dotnetcore/FreeSql#1081] [Q&A]
### 【主库】ObjectPool.Get() timeout 10 seconds

---

原因一：UnitOfWork 使用未释放，请保证程序内使用 UnitOfWork 的地方会执行 Dispose

原因二：Max Pool Size 设置过小，程序访问量过高

监视 fsql.Ado.MasterPool.Statistics，它的值：Pool: 5/100, Get wait: 0, GetAsync await: 0

```
5 为可用连接数，值为0后开始排队
100 为当前最大连接数
Get await 为同步方法获取连接的排队数量（超过10秒就会报错）
GetAsync await 为异步方法获取连接的排队数量
监视 FreeSql.UnitOfWork.DebugBeingUsed 这个静态字典，存储正在使用事务的工作单元
```

注意：尽量不要使用 fsql.Ado.MasterPool.Get() 或 GetAsync() 方法，否则请检查姿势。

**Comment by pengxinglove** [comment]
![image](https://github.com/dotnetcore/FreeSql/assets/10840137/2d9ce8df-5031-4911-b708-c600d3e54f67)
你好，我的连接数据设置100 ，运行统计还是5/5

**Reply by 2881099** [reply]
简单点，开启 ado.net 自带的连接池 UseAdoConnetionPool(true)
